lib/src/rules/dartdoc_summary_line.dart

```diff
@@ -0,0 +1,185 @@
+
+**DO** start doc comments with a single-sentence summary.
+
+Start your doc comment with a brief, user-centric description ending with a
+period. A sentence fragment is often sufficient. Provide just enough context
+for the reader to orient themselves and decide if they should keep reading or
+look elsewhere for the solution to their problem.
+
+For example:
+
+**GOOD:**
+```dart
+/// Deletes the file at [path] from the file system.
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Depending on the state of the file system and the user's permissions,
+/// certain operations may or may not be possible. If there is no file at
+/// [path] or it can't be accessed, this function throws either [IOError]
+/// or [PermissionError], respectively. Otherwise, this deletes the file.
+void delete(String path) {
+  ...
+}
+```
+
+From [Effective Dart](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph):
+
+**DO** separate the first sentence of a doc comment into its own paragraph.
+
+Add a blank line after the first sentence to split it out into its own
+paragraph. If more than a single sentence of explanation is useful, put the
+rest in later paragraphs.
+
+This helps you write a tight first sentence that summarizes the documentation.
+Also, tools like dart doc use the first paragraph as a short summary in places
+like lists of classes and members.
+
+For example:
+
+**GOOD:**
+```dart
+/// Deletes the file at [path].
+///
+/// Throws an [IOError] if the file could not be found. Throws a
+/// [PermissionError] if the file is present but could not be deleted.
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Deletes the file at [path]. Throws an [IOError] if the file could not
+/// be found. Throws a [PermissionError] if the file is present but could
+/// not be deleted.
+void delete(String path) {
+  ...
+}
+```
+''';
+
+/// Enforces that DartDoc blocks start with a single, brief line that ends in
+/// a period.
+///
+/// (Like the one above!)
+class DartdocSummaryLine extends LintRule {
+  DartdocSummaryLine()
+      : super(
+            name: 'dartdoc_summary_line',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addComment(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  /// The summary line should be 'brief' according to the Dart recommendations,
+  /// however 'brief' is not defined perfectly. Instead of enforcing a fixed
+  /// number of characters, we just look at how many lines are part of the
+  /// summary, and if we deem them to be too many (2 should be plenty), we fail
```

Comment:
I think we need some data on this point. If plenty of "fine" doc comments exist which are written across more than 2 lines, we'll have to adjust, so that the rule can be used.

Reply:
Since part of this is geared towards the generated Dartdoc... Do we have examples where the generated dartdoc looks good/bad because of the summary, and try to infer what's a good limit here? IIRC Dartdoc trimmed this at some point? (or I remember a [...] which pointed to the full doc page or something?)

If unlimited, people can write walls of text as the "first line" of the dartdoc, and that kind of defeats the purpose of the lint I think :/

Reply:
Drive by: Given that it's important for the lint and for dartdoc to agree, have we considered moving the logic that dartdoc uses to find the summary into the `markdown` package (or maybe even to the `analyzer` package as a getter on `Comment`) so that it can be shared? Having the logic in two places opens the door for them becoming inconsistent in the future.

Reply:
So how does it decide what to include in the first HTML node?

Reply:
Right. I should probably drop out of the conversation with apologies for distracting everyone. It just seemed to me that the logic dartdoc / markdown uses to determine what's in the first HTML node could be used without needing to first build the HTML and then pull it back out. But if it isn't worth the effort then it isn't.